README.md

```diff
@@ -46,11 +46,77 @@ Example:
 * "app name" - The application name (default is go-mssqldb)
 * "ApplicationIntent" - Can be given the value "ReadOnly" to initiate a read-only connection to an Availability Group listener.
 
-Example:
+The connection string can be specified in three formats:
 
-```go
-    db, err := sql.Open("mssql", "server=localhost;user id=sa")
-```
+1. `key=value` pairs separated by `;`
+
+    This format doesn't allow characters like `;` within a value. Leading and trailing whitespace is stripped.
+
+    Example:
+
+    ```go
+    db, err := sql.Open("mssql", "server=localhost;user id=sa;password=foobar")
+    ```
+
+1. ODBC format - `key=value` pairs separated by `;`. The string is prefixed with `odbc:`
+
+    This format allows characters like `;` within a value if the value is wrapped in `{}`. Leading and trailing whitespace is retained for wrapped values.
+
+    Example:
+
+    ```go
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password={foo;bar}") // password is "foo;bar"
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password={foo{bar}") // password is "foo{bar"
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password={foobar }") // password is "foobar "
+    ```
+
+    If the value is not wrapped in `{}`, then `{` or `}` characters are taken literally.
+
+    Example:
+
+    ```go
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password=foo{bar") // password is "foo{bar"
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password=foo}bar") // password is "foo}bar"
+    ```
+
+    If the value is wrapped in `{}` and needs to contain the `}` character, it can be escaped by doubling it.
+
+    Example:
+
+    ```go
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password={foo{bar}") // password is "foo{bar"
+    db, err := sql.Open("mssql", "odbc:server=localhost;user id=sa;password={foo}}bar}") // password is "foo}bar"
+    ```
+
+1. URL format with `sqlserver` protocol - `sqlserver://username:password@host/instance?param1=value&param2=value`
+
+    Example:
+
+    ```go
+    db, err := sql.Open("mssql", "sqlserver://someuser:foo%3A%2F%5C%21~%40;bar@somehost?connection+timeout=30") // password is ""foo:/\\!~@;bar""
```

Comment:
Please also include a more normal url string with fewer encodings and an instance like:
`sqlserver://dbuser:dbpass@hostname/instance?database=master&connection+timeout=30`